src/utils/WorkoutShare.tsx

```diff
@@ -0,0 +1,106 @@
+import { useContext, useEffect } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import styled from 'styled-components';
+import TimerContext from '../components/contex/TimerContext';
+
+const ShareContainer = styled.div`
+  display: flex;
+  align-items: center;
+  gap: 1rem;
+  margin-bottom: 1rem;
+`;
+
+const ShareButton = styled.button`
+  background-color: #007bff;
+  color: white;
+  padding: 10px 20px;
+  margin: 5px;
+  border: none;
+  border-radius: 8px;
+  font-size: 1.2rem;
+  cursor: pointer;
+  display: flex;
+  align-items: center;
+  gap: 8px;
+  
+  &:hover {
+    background-color: #0056b3;
+  }
+  
+  &:disabled {
+    opacity: 0.5;
+    cursor: not-allowed;
+    background-color: #007bff;
+  }
+`;
+
+const ShareIcon = styled.svg`
+  width: 20px;
+  height: 20px;
+`;
+
+const Alert = styled.div`
+  padding: 12px 20px;
+  background-color: #4caf50;
+  color: white;
+  border-radius: 8px;
+  font-size: 1rem;
+`;
+
+const WorkoutShareControls = () => {
+    const { state, dispatch } = useContext(TimerContext);
+    const [searchParams, setSearchParams] = useSearchParams();
+
+    // Load state from URL on mount
+    // biome-ignore lint/correctness/useExhaustiveDependencies: <explanation>
+    useEffect(() => {
+        const timerParam = searchParams.get('workout');
+        if (timerParam) {
+            try {
+                const decodedTimers = JSON.parse(atob(timerParam));
+
+                // Validate the structure of decoded timers
+                if (Array.isArray(decodedTimers) && decodedTimers.every(timer => timer.id && timer.type && typeof timer.duration === 'number')) {
+                    dispatch({ type: 'LOAD_TIMERS', payload: decodedTimers });
+                }
+            } catch (error) {
+                console.error('Failed to load workout from URL:', error);
+            }
+        }
+    }, []);
+
+    const handleSaveToUrl = () => {
+        if (state.timers.length === 0) {
+            return;
+        }
+
+        // Create a simplified version of timers for URL sharing
+        const shareableTimers = state.timers.map(timer => ({
+            id: timer.id,
+            type: timer.type,
+            duration: timer.duration,
+            config: timer.config,
+        }));
+
+        // Encode timers as base64 to make URL-friendly
+        const encodedTimers = btoa(JSON.stringify(shareableTimers));
+        setSearchParams({ workout: encodedTimers });
+    };
```

Comment:
how can you do this so that the user doesn't have to click the button in order to save state to URL? It would be nice to automatically just add to the URL as the user modifies the timer configurations